examples/chatgpt/gpt_actions_library/gpt_action_Notion.ipynb

```diff
@@ -0,0 +1,534 @@
+    "            application/json:\n",
+    "              schema:\n",
+    "                type: object\n",
+    "                properties:\n",
+    "                  object:\n",
+    "                    type: string\n",
+    "                  results:\n",
+    "                    type: array\n",
+    "                    items:\n",
+    "                      type: object\n",
+    "                      properties:\n",
+    "                        id:\n",
+    "                          type: string\n",
+    "                        title:\n",
+    "                          type: array\n",
+    "                          items:\n",
+    "                            type: object\n",
+    "                            properties:\n",
+    "                              type:\n",
+    "                                type: string\n",
+    "                              text:\n",
+    "                                type: object\n",
+    "                                properties:\n",
+    "                                  content:\n",
+    "                                    type: string"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Authentication Instructions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below are instructions on setting up authentication with this 3rd party application."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Pre-Action Steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before you set up authentication in ChatGPT, please take the following steps in the Notion."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Go to the Notion Settings Page for your workspace\n",
+    "2. Navigate to My Connections > Develop or Manage Integrations\n",
+    "3. Create new Integration marked as Internal\n",
+    "4. Locate your integration and find the API Key labeled: Internal Integration Secret.  This is the bearer token for this integration.\n",
+    "\n",
+    "**NOTE!** You need to share specific pages, databases, or wikis with the integration in order to access them in ChatGPT.  Do this by selecting the ... button on the upper right of a page and select the appropriate connection."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![notion_connections.png](../../../images/creating_notion_integration.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![sharing_notion_pages.png](../../../images/sharing_notion_with_GPT.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### In ChatGPT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In ChatGPT, click on \"Authentication\" and choose **\"API Key\"**. Enter in the information below. "
```

Comment:
Is OAuth possible? It's ok you chose API key but OAuth is best practice, so we at least want to mention a URL for how to set up OAuth - see the Canvas example here: https://github.com/openai/openai-cookbook/blob/main/examples/chatgpt/gpt_actions_library/gpt_action_canvaslms.ipynb